mongomock/aggregate.py

```diff
@@ -384,6 +387,66 @@ def _handle_string_operator(self, operator, values):
                 raise OperationFailure('strcasecmp must have 2 items')
             a, b = str(self.parse(values[0])), str(self.parse(values[1]))
             return 0 if a == b else -1 if a < b else 1
+        if operator == '$regexMatch':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$regexMatch expects an object of named arguments but found: %s' % type(values))
+            for field in ('input', 'regex'):
+                if field not in values:
+                    raise OperationFailure("$regexMatch requires '%s' parameter" % field)
+            unknown_args = set(values) - {'input', 'regex', 'options'}
+            if unknown_args:
+                raise OperationFailure(
+                    '$regexMatch found an unknown argument: %s' % list(unknown_args)[0])
+
+            try:
+                input_value = self.parse(values['input'])
```

Comment:
Can `KeyError` be raised here and at line 410? The keys were checked at line 395.

Reply:
The `self.parse()` is rising `KeyError` when given parameter references not existing field.

For example this will return `False` - just like real MongoDB:

```py
{"$regexMatch": {"input": "$missingField", regex: "foo"}}
```

Reply:
My bad, haven't checked the code for `self.parse()`. Thank you for the explanation @kosqx.

Reply:
Thanks @kosqx for answering :-)